scripts/talos/index.test.js

```diff
@@ -0,0 +1,111 @@
+jest.spyOn(global.console, 'log').mockImplementation(() => {});
+jest.spyOn(global.console, 'error').mockImplementation(() => {});
+
+describe('Talos', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    getChangedPackages.mockImplementation(() => ['package1', 'package2']);
+    getBranchName.mockImplementation(() => 'branchname');
+    upgradeDependencies.mockImplementation(() =>
+      Promise.resolve(['apple', 'pears']),
+    );
+    createPullRequest.mockImplementation(() =>
+      Promise.resolve({ data: { html_url: 'prURL' } }),
+    );
+    bumpPackages.mockImplementation(() => Promise.resolve());
+  });
+
+  it('should run expected commands to create PR', async () => {
+    await talos();
+
+    expect(getChangedPackages).toHaveBeenCalledTimes(1);
+
+    expect(upgradeDependencies).toHaveBeenCalledTimes(1);
+    expect(upgradeDependencies).toHaveBeenCalledWith(['package1', 'package2']);
+
+    expect(bumpPackages).toHaveBeenCalledTimes(1);
+    expect(bumpPackages).toHaveBeenCalledWith({
+      packageNames: ['apple', 'pears'],
+      version: 'patch',
+    });
+
+    expect(getPackagePath).toHaveBeenCalledTimes(2);
+    expect(getPackagePath).toHaveBeenCalledWith('apple');
+    expect(getPackagePath).toHaveBeenCalledWith('pears');
+
+    expect(runNpmInstall).toHaveBeenCalledTimes(2);
+
+    expect(bumpChangelogs).toHaveBeenCalledTimes(1);
+    expect(bumpChangelogs).toHaveBeenCalledWith({
+      changesDescription: 'Talos - Bump Dependencies',
+      packageNames: ['apple', 'pears'],
+      prLink: 'prURL',
+    });
+
+    expect(checkoutBranch).toHaveBeenCalledTimes(1);
+    expect(checkoutBranch).toHaveBeenCalledWith('branchname');
+
+    expect(commitChanges).toHaveBeenCalledTimes(2);
+    expect(commitChanges).toHaveBeenCalledWith('Talos - Update changelogs');
+    expect(commitChanges).toHaveBeenCalledWith('Talos - Bump Dependencies');
+
+    expect(createPullRequest).toHaveBeenCalledTimes(1);
+    expect(createPullRequest).toHaveBeenCalledWith({
+      branchName: 'branchname',
+      bumpedPackages: ['apple', 'pears'],
+      packages: ['package1', 'package2'],
+    });
+
+    expect(getBranchName).toHaveBeenCalledTimes(1);
+    expect(getBranchName).toHaveBeenCalled();
+
+    expect(process.exit).not.toHaveBeenCalled();
+    expect(global.console.log).not.toHaveBeenCalled();
+  });
+
+  it('should exit if no packages were published', async () => {
+    getChangedPackages.mockImplementation(() => []);
+    await talos();
+
+    expect(process.exit).toHaveBeenCalled();
+    expect(global.console.log).toHaveBeenCalledWith(
+      'No packages were published!',
+    );
+  });
+
+  it('should exit if promise chain rejects a promise', async () => {
+    // Reject a random promise in the chain
+    const error = new Error('something bad happened');
+    checkoutBranch.mockImplementation(() => Promise.reject(error));
+
+    await talos();
+
+    expect(process.exit).toHaveBeenCalled();
```

Comment:
Could specify the non-zero error code here as this is important behaviour for the pipeline.